api/hypershift/v1beta1/controlplanecomponent_types.go

```diff
@@ -0,0 +1,83 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func init() {
+	SchemeBuilder.Register(func(scheme *runtime.Scheme) error {
+		scheme.AddKnownTypes(SchemeGroupVersion,
+			&ControlPlaneComponent{},
+			&ControlPlaneComponentList{},
+		)
+		return nil
+	})
+}
+
+const (
+	// ControlPlaneComponentAvailable indicates whether the ControlPlaneComponent is available.
+	ControlPlaneComponentAvailable ConditionType = "Available"
+
+	// WaitingForDependenciesReason indicates that there are unavailable dependencies blocking the ControlPlaneComponent reconcilation.
+	WaitingForDependenciesReason string = "WaitingForDependencies"
+	// ReconciliationErrorReason indicates that there was an error during the reconcilation of the ControlPlaneComponent.
+	ReconciliationErrorReason string = "ReconciliationError"
+)
+
+// ControlPlaneComponentSpec defines the desired state of ControlPlaneComponent
+type ControlPlaneComponentSpec struct {
+}
+
+type ComponentResource struct {
+	Kind string `json:"kind"`
```

Comment:
godoc

for consistency, we should indicate that these are required if they're not optional
and if they are immutable, then also // +immutable